alica_engine/src/engine/RunnableObject.cpp

```diff
@@ -149,4 +176,36 @@ void RunnableObject::traceInit(const std::string& type)
     }
 }
 
+void RunnableObject::setBlackboardBlueprint(const BlackboardBlueprint* blackboard)
+{
+    _blackboardBlueprint = blackboard;
+}
+
+void RunnableObject::setInput(const Blackboard* parent_bb, const KeyMapping* keyMapping)
+{
+    const auto lockedParentBb = LockedBlackboardRO(*parent_bb);
+    auto& childBb = _blackboard->impl(); // Child not started yet, no other user exists, dont' use lock
+    for (const auto& [parentKey, childKey] : keyMapping->getInputMapping()) {
+        try {
+            childBb.set(childKey, lockedParentBb.get(parentKey));
+            ALICA_DEBUG_MSG("passing " << parentKey << " into " << childKey);
+        } catch (std::exception& e) {
+            ALICA_WARNING_MSG("Blackboard error passing " << parentKey << " into " << childKey << ". " << e.what());
+        }
```

Comment:
This and output would be really nice information to put onto trace. Please create a new pr with the trace change once this is merged.

Reply:
New task https://www.wrike.com/open.htm?id=840847670